### ISSUE TITLE
Package dependency view is not updating when Package.swift updates

### ISSUE THREAD

**Issue by adam-fowler** [swift-server/vscode-swift#65]
The resolve is run but when the task finishes `PackageDependenciesProvider` does not recognise the task. It seems as if vscode has messed with the definition when you run it. The type gets changed and the definition arguments are undefined

**Comment by adam-fowler** [comment]
#66 would fix this

**Comment by svanimpe** [comment]
That's really weird. I did some digging around and found out why this happens:

In **commands.ts**, if we modify the fetched task:

```typescript
task.presentationOptions = {
  reveal: vscode.TaskRevealKind.Silent
};
```

it seems like VS Code creates a new task, with a new definition. If I instead set the `presentationOptions` when creating a resolve task in the provider, the definition is unmodified. This is probably also why in `SwiftTaskProvider.resolveTask` I had to create a new `Task` object, and was unable to just "fill in" the tasked passed in as a parameter.

So yeah, seems like we really need to use a task's name to identify it, not its ID or definition.
